Firmware/ConfigurationStore.cpp

```diff
@@ -283,16 +282,13 @@ bool Config_RetrieveSettings()
     else
     {
         Config_ResetDefault();
-    //Return false to inform user that eeprom version was changed and firmware is using default hardcoded settings now.
-    //In case that storing to eeprom was not used yet, do not inform user that hardcoded settings are used.
-    if (eeprom_read_byte(reinterpret_cast<uint8_t*>(&(EEPROM_M500_base->version[0]))) != 0xFF ||
-      eeprom_read_byte(reinterpret_cast<uint8_t*>(&(EEPROM_M500_base->version[1]))) != 0xFF ||
-      eeprom_read_byte(reinterpret_cast<uint8_t*>(&(EEPROM_M500_base->version[2]))) != 0xFF)
-    {
-      previous_settings_retrieved = false;
-    }
+        //Return false to inform user that eeprom version was changed and firmware is using default hardcoded settings now.
+        //In case that storing to eeprom was not used yet, do not inform user that hardcoded settings are used.
+        if (eeprom_is_initialized_block(EEPROM_M500_base->version, sizeof(EEPROM_M500_base->version))) {
```

Comment:
`sizeof(EEPROM_M500_base->version) != 3` but 4, I think this doesn't match the previous implementation. Not sure though if it matters.

Reply:
May be: if the EEPROM chip is empty, version at the memory cells will be just 4 bytes od 0xff, so the check will probably work ok.

Reply:
That was my thought as well. An eeprom that is empty for real will have all cells 0xFF. If any of those cells are not 0xFF, then it means something was stored there. If the data is garbage, then the Version string will mismatch and the default settings will be loaded